### ISSUE TITLE
cppcheck: assorted null pointer suspects 

### ISSUE THREAD

**Issue by chipitsine** [ppp-project/ppp#390]
```
chat/chat.c:621:15: warning: Either the condition 'rep_len+1<=sizeof(report_buffer)' is redundant or the array 'report_buffer[256]' is accessed at index 256, which is out of bounds. [arrayIndexOutOfBoundsCond]
chat/chat.c:230:13: warning: Either the condition '!ans' is redundant or there is possible null pointer dereference: ans. [nullPointerRedundantCheck]
chat/chat.c:397:32: warning: Either the condition 'cfp==NULL' is redundant or there is possible null pointer dereference: cfp. [nullPointerRedundantCheck]
chat/chat.c:1159:53: warning: Either the condition 'f==NULL' is redundant or there is possible null pointer dereference: f. [nullPointerRedundantCheck]
chat/chat.c:247:12: warning: Either the condition '!s' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/auth.c:602:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:603:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:604:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:623:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:624:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:625:5: warning: Either the condition 'wp==NULL' is redundant or there is possible null pointer dereference: wp. [nullPointerRedundantCheck]
pppd/auth.c:2313:6: warning: Either the condition 'ap==NULL' is redundant or there is possible null pointer dereference: ap. [nullPointerRedundantCheck]
pppd/auth.c:2314:13: warning: Either the condition 'ap==NULL' is redundant or there is possible null pointer dereference: ap. [nullPointerRedundantCheck]
pppd/auth.c:602:29: warning: Either the condition 'wp==NULL' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/auth.c:623:29: warning: Either the condition 'wp==NULL' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/auth.c:2313:30: warning: Either the condition 'ap==NULL' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/auth.c:2651:6: warning: Either the condition 'ap==NULL' is redundant or there is possible null pointer dereference: ap. [nullPointerRedundantCheck]
pppd/auth.c:2652:13: warning: Either the condition 'ap==NULL' is redundant or there is possible null pointer dereference: ap. [nullPointerRedundantCheck]
pppd/auth.c:2651:30: warning: Either the condition 'ap==NULL' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/eap-tls.c:741:5: warning: Either the condition '!esp->es_client.ea_session' is redundant or there is possible null pointer dereference: ets. [nullPointerRedundantCheck]
pppd/main.c:1365:5: warning: Either the condition '(newp=(struct callout*)malloc(sizeof(struct callout)))==NULL' is redundant or there is possible null pointer dereference: newp. [nullPointerRedundantCheck]
pppd/main.c:1366:5: warning: Either the condition '(newp=(struct callout*)malloc(sizeof(struct callout)))==NULL' is redundant or there is possible null pointer dereference: newp. [nullPointerRedundantCheck]
pppd/main.c:1368:5: warning: Either the condition '(newp=(struct callout*)malloc(sizeof(struct callout)))==NULL' is redundant or there is possible null pointer dereference: newp. [nullPointerRedundantCheck]
pppd/main.c:1369:5: warning: Either the condition '(newp=(struct callout*)malloc(sizeof(struct callout)))==NULL' is redundant or there is possible null pointer dereference: newp. [nullPointerRedundantCheck]
pppd/main.c:1370:9: warning: Either the condition '(newp=(struct callout*)malloc(sizeof(struct callout)))==NULL' is redundant or there is possible null pointer dereference: newp. [nullPointerRedundantCheck]
pppd/main.c:2062:2: warning: Either the condition 'np==0' is redundant or there is possible null pointer dereference: np. [nullPointerRedundantCheck]
pppd/main.c:2063:2: warning: Either the condition 'np==0' is redundant or there is possible null pointer dereference: np. [nullPointerRedundantCheck]
pppd/main.c:2064:2: warning: Either the condition 'np==0' is redundant or there is possible null pointer dereference: np. [nullPointerRedundantCheck]
pppd/options.c:1000:5: warning: Either the condition 'list==0' is redundant or there is possible null pointer dereference: list. [nullPointerRedundantCheck]
pppd/options.c:1001:5: warning: Either the condition 'list==0' is redundant or there is possible null pointer dereference: list. [nullPointerRedundantCheck]
pppd/options.c:658:19: warning: Either the condition 'path==NULL' is redundant or there is pointer arithmetic with NULL pointer. [nullPointerArithmeticRedundantCheck]
pppd/peap.c:481:2: warning: Either the condition '!psm' is redundant or there is possible null pointer dereference: psm. [nullPointerRedundantCheck]
pppd/peap.c:482:7: warning: Either the condition '!psm' is redundant or there is possible null pointer dereference: psm. [nullPointerRedundantCheck]
pppd/plugins/radius/avpair.c:440:7: warning: Either the condition 'this_node!=(struct value_pair*)NULL' is redundant or there is possible null pointer dereference: this_node. [nullPointerRedundantCheck]
pppd/plugins/radius/avpair.c:441:2: warning: Either the condition 'this_node!=(struct value_pair*)NULL' is redundant or there is possible null pointer dereference: this_node. [nullPointerRedundantCheck]
pppd/tls.c:243:9: warning: Either the condition '!tmp' is redundant or there is possible null pointer dereference: tmp. [nullPointerRedundantCheck]
pppd/tty.c:421:29: warning: Either the condition 'n>0xFF' is redundant or there is signed integer overflow for expression '1<<(n&0x1F)'. [integerOverflowCond]
pppdump/zlib.c:735:46: warning: Either the condition 'strm==0' is redundant or there is possible null pointer dereference: strm. [nullPointerRedundantCheck]
pppdump/zlib.c:857:46: warning: Either the condition 'strm==0' is redundant or there is possible null pointer dereference: strm. [nullPointerRedundantCheck]
```

**Comment by enaess** [comment]
Glancing over this, none of these seems particularly interesting to fix.

**Comment by paulusmack** [comment]
Most of these are because the checker doesn't know that fatal() never returns, and hence novm() also never returns.

The first chat.c report is valid, as are the avpair.c reports. The pppdump reports are valid but in unused code.

The tty.c report is technically correct in that 1 << 31 will cause integer overflow on a 32-bit system, but that is not actually related to the n > 0xFF test.

**Comment by paulusmack** [comment]
Fixed by 3b0fe68ef1e2, acc6cd4aab0a, efb2667c2529 and f2d849bd46d5.
